Core/include/Acts/Propagator/DirectNavigator.hpp

```diff
@@ -94,17 +93,21 @@ class DirectNavigator {
   struct State {
     /// Externally provided surfaces - expected to be ordered
     /// along the path
-    SurfaceSequence surfaceSequence = {};
+    SurfaceSequence navSurfaces = {};
 
     /// Iterator the the next surface
-    SurfaceIter nextSurfaceIter = surfaceSequence.begin();
+    SurfaceIter navSurfaceIter = navSurfaces.begin();
 
     /// Navigation state - external interface: the start surface
     const Surface* startSurface = nullptr;
     /// Navigation state - external interface: the current surface
     const Surface* currentSurface = nullptr;
     /// Navigation state - external interface: the target surface
     const Surface* targetSurface = nullptr;
+    /// Navigation state - starting layer
+    const Layer* startLayer = nullptr;
+    /// Navigation state - target layer
+    const Layer* targetLayer = nullptr;
```

Comment:
Are the `startLayer` and `targetLayer` used somewhere?

Reply:
They are called by the KF Actor and checked whether or not they are a `nullptr` to determine the volume, I believe [here](https://github.com/acts-project/acts/blob/73ea350cf349d5d9394d02ae4faaf85dfdef21a6/Core/include/Acts/TrackFitting/KalmanFitter.hpp#L410). For directed surfaces I don't think this makes sense though, since the surfaces may not be a part of the volume.